deps/media-playback/media-playback/media.h

```diff
@@ -38,11 +38,66 @@ extern "C" {
 #pragma warning(pop)
 #endif
 
+
 typedef void (*mp_video_cb)(void *opaque, struct obs_source_frame *frame);
 typedef void (*mp_audio_cb)(void *opaque, struct obs_source_audio *audio);
 typedef void (*mp_stop_cb)(void *opaque);
+typedef bool (*mp_state_handler)(struct mp_media_t *m);
+
+typedef struct mp_media_queue_node mp_media_queue_node_t;
+
+
+/*Used internally by mp_media to track current state.
+  to add new event:
+  1. Declare event in mp_media_state enumerator
+  2. Define event handler function.
+  3. Register event handler reference to queue.
+  4. Register any transition actions to the state change method.
+  Then use an invoke function to push the message to queue.
+  Note: States are mapped to an integer through a 2^x function,
```

Comment:
Is there a purpose behind using a magic function (that's not actually 2^x as evidenced by the code that does this mapping) instead of assigning the appropriate values directly in the enum?

Reply:
I see your point. My thinking was that it would make implementing new states easier and avoid potential issues from typos, but this is adding a lot of computations for a management item that is not particularly difficult or complex.